src/ui/public/saved_object_table/item_table_state.js

```diff
@@ -0,0 +1,38 @@
+import { SortOrder } from 'ui_framework/components/table/sort_order';
+
+export class ItemTableState {
```

Comment:
> What @cjcenizal is essentially proposing is to turn the ItemSelector from a collection of actions into something very similar to a flux store that handles the selection state.

Well, I'd like to steer the conversation clear of Flux and Redux for the moment. I'd really like to focus on applying **sound OO design principles**, e.g. abstraction, encapsulation, SOLID. I just want to simplify the problem to "designing software well" without being distracted by a particular state management system (for now).

> So we're getting into the "pattern for state management" discussion again.

I really want to avoid this discussion. It's an important one but I don't think this is the right time for it. I think we'll have a more constructive discussion if we focus on defining helpful abstractions, encapsulating the right implementation details, and designing the right interfaces.

Like I said... let's make this a gradual process. If we discover we need to explore patterns for state management, then let's do that, but only when we need to.

> I would consider "single state" and separate actions to be the safer, more future-proof pattern (by virtue of being just a special case that can be represented in most state management implementations I've seen).

Yes (but 😄 )... I think simple, easy-to-understand code is the ultimate form of future-proofing. I think we should not engineer for a particular future case. Instead, I'd really like to focus on making this code simple, easy to understand, and easy to refactor in the future. This way we're not cutting ourselves off from adopting Redux or Flux, but we're also not committed to a particular future path either.

Would you be OK with trying out this approach, and exploring data-flow and state-mangement patterns in a subsequent iteration?